hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -0,0 +1,312 @@
+        round.forEachEventTransaction((event, txn) -> handlePlatformTransaction(state, event, txn));
+    }
+
+    private void handlePlatformTransaction(
+            @NonNull final HederaState state,
+            @NonNull final ConsensusEvent platformEvent,
+            @NonNull final ConsensusTransaction platformTxn) {
+        // skip system transactions
+        if (platformTxn.isSystem()) {
+            return;
+        }
+
+        // Get the consensus timestamp
+        final Instant consensusNow = platformTxn.getConsensusTimestamp();
+
+        // Setup record builder list
+        recordManager.startUserTransaction(consensusNow);
+        final var recordBuilder = new SingleTransactionRecordBuilder(consensusNow);
+        final var recordListBuilder = new RecordListBuilder(recordBuilder);
+
+        // Setup configuration
+        var configuration = configProvider.getConfiguration();
+
+        try {
+            final var verifications = getVerifications(state, platformEvent, platformTxn, configuration);
+
+            // Read all signature verifications. This will also wait, if validation is still ongoing.
+            final var keyVerifications = new HashMap<Key, SignatureVerification>();
+            for (final var entry : verifications.keyVerifications().entrySet()) {
+                // TODO: Implement timeout
+                final var verification = entry.getValue().get();
+                if (verification.failed()) {
+                    throw new HandleException(ResponseCodeEnum.INVALID_SIGNATURE);
+                }
+                keyVerifications.put(entry.getKey(), verification);
+            }
+
+            // Setup context
+            final var txBody = verifications.txBody();
+            final var stack = new SavepointStackImpl(state, configuration);
+            final var verifier = new HandleContextVerifier(keyVerifications);
+            final var context = new HandleContextImpl(
+                    txBody,
+                    TransactionCategory.USER,
+                    recordBuilder,
+                    stack,
+                    verifier,
+                    recordListBuilder,
+                    checker,
+                    dispatcher,
+                    serviceScopeLookup);
+
+            // Dispatch the transaction to the handler
+            dispatcher.dispatchHandle(context);
+            configuration = configProvider.getConfiguration();
+            stack.configuration(configuration);
+
+            // TODO: Kick off special file handling if needed
+
+            // TODO: Finalize transaction with the help of the token service
+
+            // commit state
+            stack.commit();
+        } catch (PreCheckException e) {
+            recordFailedTransaction(e.responseCode(), recordBuilder, recordListBuilder);
+        } catch (HandleException e) {
+            recordFailedTransaction(e.getStatus(), recordBuilder, recordListBuilder);
+        } catch (Throwable e) {
```

Comment:
Two issues here:
1. Catching Throwable means you're going to catch, _and ignore_ fatal errors like out of memory error.  Consider catching RuntimeException instead.
1. There is an InterruptedException being hidden here, and you need to re-interrupt (`Thread.currentThread().interrupt()`) when catching that exception to ensure that the software will exit cleanly on system interrupt (Signal 15).  This is another reason to catch RuntimeException here and catch the checked exceptions individually.

Reply:
Fixed #2. For #1, there is a tradeoff here. I agree that it is a problem for some types of exceptions, but we have seen other throwables in the past (granted, that SHOULDN'T have been there, and were removed, but did show up). I want to make sure if any random throwable happens, we don't fail uncontrollably. Maybe we should carve out some additional exceptions like OOM? I'm not sure. In any even, I'm going to merge it as is, and we can come back to this debate as we fine tune. We should do the same thing in preHandleWorkflowImpl, which is also handling Throwable right now IIRC.